From 64e4958e61f2c485a8a457fbfc939b5901ae8613 Mon Sep 17 00:00:00 2001
From: Ioana Ciornei <ioana.ciornei@nxp.com>
Date: Thu, 19 May 2016 18:14:05 +0300
Subject: [PATCH 014/388] fman: check return value of
 FmPcdPlcrGetAbsoluteIdByProfileParams

Check return value of FmPcdPlcrGetAbsoluteIdByProfileParams and take
appropriate measures in case it occurs an error

Signed-off-by: Ioana Ciornei <ioana.ciornei@nxp.com>
[Original patch taken from SDK-V2.0-1703]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Pcd/fm_cc.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Pcd/fm_cc.c b/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Pcd/fm_cc.c
index 84d1d05..a9ac459 100644
--- a/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Pcd/fm_cc.c
+++ b/drivers/net/ethernet/freescale/sdk_fman/Peripherals/FM/Pcd/fm_cc.c
@@ -2175,6 +2175,7 @@ static void FillAdOfTypeResult(t_Handle h_Ad,
     uint32_t tmp = 0, tmpNia = 0;
     uint16_t profileId;
     t_Handle p_AdNewPtr = NULL;
+    t_Error err = E_OK;
 
     /* There are 3 cases handled in this routine of building a "result" type AD.
      * Case 1: No Manip. The action descriptor is built within the match table.
@@ -2287,12 +2288,15 @@ static void FillAdOfTypeResult(t_Handle h_Ad,
                     if (p_CcNextEngineParams->params.plcrParams.sharedProfile)
                     {
                         tmpNia |= NIA_PLCR_ABSOLUTE;
-                        FmPcdPlcrGetAbsoluteIdByProfileParams(
+                        err = FmPcdPlcrGetAbsoluteIdByProfileParams(
                                 (t_Handle)p_FmPcd,
                                 e_FM_PCD_PLCR_SHARED,
                                 NULL,
                                 p_CcNextEngineParams->params.plcrParams.newRelativeProfileId,
                                 &profileId);
+			if (err != E_OK)
+				RETURN_ERROR(MAJOR, err, NO_MSG);
+
                     }
                     else
                         profileId =
-- 
2.9.3

